ehrapy/tools/cohort_tracking/_cohort_tracker.py

```diff
@@ -336,13 +374,49 @@ def plot_cohort_barplot(
         # These list of lists is needed to reverse the order of the legend labels,
         # making the plot much more readable
         legend_handles.reverse()
-        legend_handels = [item for sublist in legend_handles for item in sublist]
 
         tot_legend_kwargs = {"loc": "best", "bbox_to_anchor": (1, 1)}
         if legend_kwargs is not None:
             tot_legend_kwargs.update(legend_kwargs)
 
-        plt.legend(handles=legend_handels, **tot_legend_kwargs)
+        def create_legend_with_subtitles(patches_list, subtitles_list, tot_legend_kwargs, legend_subtitles_bold=True):
```

Comment:
this expects such a specific structure I thought its better to not put it out of the method, to not tempt anyone to re-use this in another context^^